oak_functions/wasm/src/lib.rs

```diff
@@ -116,70 +127,11 @@ where
     }
 }
 
-/// `WasmState` holds runtime values for a particular execution instance of Wasm, handling a
-/// single user request. The methods here correspond to the ABI OAK_FUNCTIONS functions that allow
-/// the Wasm module to exchange the request and the response with the Oak functions server. These
-/// functions translate values between Wasm linear memory and Rust types.
-// TODO(#3796): Remove WasmState.
-pub struct WasmState<L: OakLogger> {
-    instance: wasmi::Instance,
-    store: wasmi::Store<UserState<L>>,
-}
-
-impl<L> WasmState<L>
-where
-    L: OakLogger,
-{
-    pub fn new(
-        module: Arc<wasmi::Module>,
-        request_bytes: Vec<u8>,
-        logger: L,
-        extensions: HashMap<ExtensionHandle, Box<dyn OakApiNativeExtension>>,
-    ) -> anyhow::Result<Self> {
-        let user_state = UserState::init(request_bytes, extensions, logger);
-        // For isolated requests we need to create a new store for every request.
-        let mut store = wasmi::Store::new(module.engine(), user_state);
-        let linker = OakLinker::new(module.engine(), &mut store);
-        let (instance, store) = linker.instantiate(store, module)?;
-
-        instance.exports(&store).for_each(|export| {
-            store
-                .data()
-                .logger
-                .log_sensitive(Level::Info, &format!("instance exports: {:?}", export))
-        });
-
-        let wasm_state = Self { instance, store };
-        Ok(wasm_state)
-    }
-
-    // Invokes the Wasm module by calling main.
-    fn invoke(&mut self) {
-        let main = self
-            .instance
-            .get_typed_func::<(), ()>(&self.store, MAIN_FUNCTION_NAME)
-            .expect("couldn't get `main` export");
-        let result = main.call(&mut self.store, ());
-        self.store.data().logger.log_sensitive(
-            Level::Info,
-            &format!("running Wasm module completed with result: {:?}", result),
-        );
-    }
-
-    // Needed for unit tests.
-    #[allow(dead_code)]
-    fn get_request_bytes(&self) -> Vec<u8> {
-        let user_state = self.store.data();
-        user_state.request_bytes.clone()
-    }
-
-    fn get_response_bytes(&self) -> Vec<u8> {
-        let user_state = self.store.data();
-        user_state.response_bytes.clone()
-    }
-}
-
-// Exports the functions from oak_functions_abi/src/lib.rs.
+/// The `OakLinker` exports the functions from the ABI of Oak Functions. These functions allow the
```

Comment:
```suggestion
/// Exports the functions from the ABI of Oak Functions. These functions allow the
```

No need to repeat the name of the documented class.